### DISCUSSION TITLE
Unable to delete

### DISCUSSION THREAD

**Discussion by mmol67** [immich-app/immich#3192] [Q&A]
Immich 1.66.1, Android app ver. 1.65.0
Xiaomi Redmi Note 9
App has all permissions. It was working some days ago.

What is happening to me:

-  deleting in Immich Android app: the picture is deleted in the app, and in the server, but not in the gallery, so it reappears in the app, and in the server after a few seconds, first only cloud icon, then cloud icon with the check.
- deleting in Immich web app: same thing, it reappears after a few seconds.
- deleting in phone gallery: then the image icon changes to only cloud, then delete image in the app, and now the image dissapears (at last!)

So it seems that the Android app is unable to delete the image from gallery in phone storage, and this causes this behaviour.

Is this a bug or can I fix it doing something?

**Comment by alextran1502** [comment]
If you have foreground backup enabled and delete the photo just from the Immich server, not on your phone's native library, the file will be uploaded again.

**Reply by cibernox** [reply]
It depends on wether we want to make immich the authoritative source of truth for the photos. If the user wants (should they be given the option?) deleing an app from immich should delete it from the device it came from too.

**Reply by mmol67** [reply]
Actually, the problem is that if you delete a picture in the Immich Android app, it does not get removed from the phone, so it reupload this picture again. This was not happening before.

**Reply by cibernox** [reply]
FWIW, I was referring to the web version. The iOS app does ask you (annoyingly) when you delete a picture if you want to delete also from the iPhones gallery.

**Comment by mmol67** [comment]
Hello again. The thing is that when you delete a picture in the android app, it shows a dialog saying: Delete Permanently. These items will be permanently deleted from Immich and from your device. And then ask for permission to send images to trash. Then "Allow". The images dissapear, from Immich Android app and from the server (web app). 

However they remain in gallery, and after some seconds, they reappear in the Immich Android app and in the server. So Immich Android app is not deleting the pictures from the phone, as it says.

EDIT: if you don't open the Immich Android App, the images keep deleted in the server. When you open again the Immich Android app, the images appear in this app, and then in the server. This is a bug. For sure.
 
![image](https://github.com/immich-app/immich/assets/60692795/22c78505-7861-4ea1-80de-afb8c9276b95)
![image](https://github.com/immich-app/immich/assets/60692795/c3e5ebfb-6ca1-44e0-a5c3-52aef99d46c2)
